cmd/oras/internal/option/pusher.go

```diff
@@ -18,26 +18,42 @@ package option
 import (
 	"context"
 	"encoding/json"
+	"errors"
+	"fmt"
 	"os"
+	"regexp"
+	"strings"
 
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/spf13/pflag"
 	"oras.land/oras-go/v2/content"
 )
 
+const (
+	annotationManifest = "$manifest"
+)
+
+var (
+	errAnnotationConflict    = errors.New("annotation input conflict")
+	errAnnotationFormat      = errors.New("annotation MUST be a key-value pair")
+	errAnnotationDuplication = errors.New("annotation key duplication")
```

Comment:
Since those errors need to be checked and thrown in a fail-fast way. I suggest 1) add `annotations map[string]map[string]string` to the `Pusher` struct; 2) Make sure `LoadManifestAnnotations` get called in `cobra.Command.PreRunE`